src/herder/HerderImpl.cpp

```diff
@@ -861,16 +860,20 @@ HerderImpl::sendSCPStateToPeer(uint32 ledgerSeq, Peer::pointer peer)
     // ledger to achieve this
     if (delayCheckpoint)
     {
-        mEarlyCatchupTimer.expires_from_now(
-            Herder::SEND_LATEST_CHECKPOINT_DELAY);
-        mEarlyCatchupTimer.async_wait(
-            [checkpoint, this, sendSlot]() {
-                getSCP().processCurrentState(
-                    checkpoint,
-                    [&](SCPEnvelope const& e) { return sendSlot(e, true); },
-                    false);
-            },
-            &VirtualTimer::onFailureNoop);
+        auto weakPeer = std::weak_ptr<Peer>(peer);
```

Comment:
It looks like we have redundant peer pointers. If you lock the weak pointer here, you should modify `sendSlot` to take `peerPtr` as an argument instead of creating and locking another weak_ptr again for the same object.

Reply:
Just replaced with a using `peer`.